core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -62,6 +62,8 @@ class Partition(val topic: String,
   private val leaderIsrUpdateLock = new ReentrantReadWriteLock
   private var zkVersion: Int = LeaderAndIsr.initialZKVersion
   @volatile private var leaderEpoch: Int = LeaderAndIsr.initialLeaderEpoch - 1
+  // defined when this broker is leader for partition
```

Comment:
nit: It initially wasn't clear to me what this variable represented. In addition to saying when it's defined, maybe add a brief description of what it means.